astropy/table/table.py

```diff
@@ -33,6 +33,29 @@
 from . import conf
 
 
+_implementation_notes = """
```

Comment:
Just to check, what is the motivation for this as opposed to comments?

Reply:
This reminds me of your (@astrofrog's) old suggestion of using the module docstring for information for developers. I liked the idea then, and I think this note shows that it certainly makes sense to have *some* general mechanism for doing that.

@taldcroft - I recall that at the time you felt that information for developers should also be in the documentation - I think there is certainly an argument for that too, but maybe your choice here suggests that having something *inside* the module is the easier route after all.

... ... ...

Actually, I don't recall quite right - it is issue #4821 that suggested adding a "README" type thingie.

For now, my sense is to leave things as they are here, but open a new issue - I would love for other modules to have similar implementation notes!

Reply:
See #8930 for a place to discuss

Reply:
The motivation was to remind myself upfront of global gotchas that could be missed if buried in the code.  Of course this could be a comment block.  Hopefully before release this will be in some better form.